app/routes/user.credentials/client_credentials.server.ts

```diff
@@ -190,15 +196,17 @@ export class ClientCredentialVendingMachine {
   }
 
   async #deleteClientCredentialInternal(client_id: string) {
-    const command = new DeleteUserPoolClientCommand({
-      ClientId: client_id,
-      UserPoolId: process.env.COGNITO_USER_POOL_ID,
-    })
+    if (await checkAppClientExists(client_id)) {
```

Comment:
Don't check that it exists before deleting it. Just delete it, and if it returns an error that indicates that it didn't exist, discard the error.